lib/plug/csrf_protection.ex

```diff
@@ -103,17 +103,34 @@ defmodule Plug.CSRFProtection do
 
   def init(opts), do: opts
 
-  def call(conn, _opts) do
+  def call(conn, opts) do
+    opts = verify_opts opts
     csrf_token = get_session(conn, "_csrf_token")
     Process.put(:plug_csrf_token, csrf_token)
-
     if not verified_request?(conn, csrf_token) do
-      raise InvalidCSRFTokenError
+      if opts[:with] == :exception do
```

Comment:
@mlnkrish didn't notice your post right away! But yeah, if we're going to support more than just 2 options like Rails does, it would make sense to use `opts[:with]`. 

If you decide to stick with `opts[:with]`, you could implement @lexmag's suggestions and then add docs. Once ready, you could ping josevalim to review/merge your PR.